### ISSUE TITLE
UnsovedSymbolException when resolving "var" variable in 3.19.0

### ISSUE THREAD

**Issue by kris-scheibe** [javaparser/javaparser#3088]
When resolving the operator `o1` in the snippet below by calling `calculateResolvedType()` on the `NameExpr` an `UnresolvedSymbolException` is thrown:

```
class C {
    boolean f() {
        var o1 = new Object();
        var o2 = new Object();

        return o1 == o2;
    }
}
```

```
Unsolved symbol : var
UnsolvedSymbolException{context='null', name='var', cause='null'}
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.convertToUsage(JavaParserFacade.java:670)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.convert(JavaParserFacade.java:728)
	at com.github.javaparser.symbolsolver.javaparsermodel.declarations.JavaParserSymbolDeclaration.getType(JavaParserSymbolDeclaration.java:163)
	at com.github.javaparser.symbolsolver.model.resolution.Value.from(Value.java:45)
	at com.github.javaparser.symbolsolver.core.resolution.Context.solveSymbolAsValue(Context.java:139)
	at com.github.javaparser.symbolsolver.resolution.SymbolSolver.solveSymbolAsValue(SymbolSolver.java:78)
	at com.github.javaparser.symbolsolver.resolution.SymbolSolver.solveSymbolAsValue(SymbolSolver.java:83)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:527)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:99)
	at com.github.javaparser.ast.expr.NameExpr.accept(NameExpr.java:79)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getTypeConcrete(JavaParserFacade.java:551)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:398)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:380)
	at com.github.javaparser.symbolsolver.JavaSymbolSolver.calculateType(JavaSymbolSolver.java:291)
	at com.github.javaparser.ast.expr.Expression.calculateResolvedType(Expression.java:568)
...
```

Test-case that still works on 3.18.0:
```
@Test
public void resolveVarTest() {
    // Setup symbol solver
    StaticJavaParser.getConfiguration()
            .setLanguageLevel(ParserConfiguration.LanguageLevel.JAVA_11)
            .setSymbolResolver(
                    new JavaSymbolSolver(new ReflectionTypeSolver())
            );

    // Setup source code
    String sourceCode = "class C {\n"
            + "    boolean f() {\n"
            + "        var o1 = new Object();\n"
            + "        var o2 = new Object();\n"
            + "\n"
            + "        return o1 == o2;\n"
            + "    }\n"
            + "}";
    CompilationUnit cu = StaticJavaParser.parse(sourceCode);

    final var equalsExpr = cu.findFirst(BinaryExpr.class);// o1 == o2

    equalsExpr.get().getLeft().calculateResolvedType(); // should resolve to java.lang.Object
}
```

**Comment by jlerbsc** [comment]
Note: I'm not sure but it seems that the type of the VariableDeclarator is a ClassOrInterfaceType instead a VarType. Not sure if this is an issue in the parser configuration or if we have to handle this case in JavaParserFacade.convertToUsage(). To investigate.

**Comment by corradomio** [comment]
"var" is part of project Lombok.
https://projectlombok.org/
https://github.com/rzwitserloot/lombok

but other similar projects are available

https://github.com/google/auto/tree/master/value

**Comment by jlerbsc** [comment]
Note: It seems to be related to an experimental feature introduced in lombok 1.16.12
https://projectlombok.org/features/var

**Comment by jlerbsc** [comment]
I am closing this issue as it seems to be resolved in the current version (3.25.0)
